### DISCUSSION TITLE
is there a security issue in the storage management example provided by supabase?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15554] [Questions]
in the example provided by supabase, the delete_storage_object is created in the public space, which means it can be called via rpc.

the pg extension uses the service role key to do the deletion, meaning it can bypass all the rls checks and the function security level is "definer". if users know urls of storage files and the existence of this rpc, they can directly rpc call this method and delete whatever files they want?

am i missing sth here or is it a real risk?

https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs#create-an-upload-widget
![image](https://github.com/supabase/supabase/assets/133625693/da00b3f4-b535-4eee-8fc1-4d8d9bfd7c04)

**Comment by GaryAustin1** [comment]
Turned this into an issue in supabase/supabase.
